### PR TITLE
CMakeLists and refactor codes

### PR DESCRIPTION
**Pull request by justbaominh**
Added base CMakeLists for project
Shortened names of GUI and console apps

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  option(USE_PCH "Use precompiled headers" ON)
+endif()
+
+project(ImageConverter)
+
+set(CMAKE_PREFIX_PATH "C:/QT_new/6.5.2/msvc2019_64")
+
+
+# Find the QtWidgets package from Qt6
+find_package(Qt6 COMPONENTS Core Gui Widgets Quick REQUIRED)
```
Comment (ThiDinh21) [inline CMakeLists.txt:14]: Remove Widget as we don't use it

File: TUI/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+set(PROJECT_NAME ImageConverterGUI)
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: Should be ImageConverterTUI

File: TUI/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+set(CMAKE_AUTOUIC ON)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+
+project(${PROJECT_NAME})
+
+# Find necessary Qt modules
+find_package(Qt6 COMPONENTS Core Quick REQUIRED)
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: We don't use Quick in TUI

File: TUI/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# # Use file(GLOB ...) to automatically pick up new source files in the future
+# file(GLOB SOURCES
+#     *.cpp
+#     *.qml
+#     *.qrc
+# )
+
+add_executable(ImageConverterGUI ${SOURCES} main.qml main.cpp qml.qrc)
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: There should be no qml files in TUI CMake

File: TUI/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+set_target_properties(${PROJECT_NAME} PROPERTIES AUTORCC ON)
+
+# Link with the required Qt modules
+target_link_libraries(ImageConverterGUI PRIVATE Qt6::Core Qt6::Quick)
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: Unlink Quick

File: TUI/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+# Enable QML debugging if required
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: Remove unnecessary comments

**Review by ThiDinh21 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (justbaominh) [reply CMakeLists.txt:14]: Will be use in the future. Will keep it here

File: TUI/CMakeLists.txt [hunk 7]
```diff
@@ -1,27 +1,13 @@
...
 
-set(CMAKE_AUTOUIC ON)
-set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTORCC ON)
+project(ImageConverterConsole)
 
-project(${PROJECT_NAME})
-
-# Find necessary Qt modules
 find_package(Qt6 COMPONENTS Core Quick REQUIRED)
```
Comment (ThiDinh21) [inline TUI/CMakeLists.txt]: Remove Quick

**Review by ThiDinh21 — APPROVED** [review]

**Pull request merged**